source/common/rds/route_config_provider_manager.cc

```diff
@@ -80,8 +81,39 @@ RouteConfigProviderSharedPtr RouteConfigProviderManager::addDynamicProvider(
     std::function<
         std::pair<RouteConfigProviderSharedPtr, const Init::Target*>(uint64_t manager_identifier)>
         create_dynamic_provider) {
+
   // RdsRouteConfigSubscriptions are unique based on their serialized RDS config.
-  const uint64_t manager_identifier = MessageUtil::hash(rds);
+  uint64_t manager_identifier;
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.normalize_rds_provider_config")) {
+    Protobuf::ReflectableMessage reflectable_message = createReflectableMessage(rds);
```

Comment:
If it is possible to change the method signature so it will be a templated method (for the `rds` argument), I think it will get around the reflection and the code will be able to access the `config_source` field directly.

Reply:
Yes, it is possible. But in order to "reset" initial_fetch_timeout, I will need to make a copy of entire config (as it was done before trying reflection). But I guess it is better than removing constness across multiple levels of calls. I will push template version soon. Thanks for great idea.

Reply:
IMHO removing the costness is OK (as long as the function also reinstates the removed timeout), but the reflection is less desirable.

Reply:
I was not sure how expensive reflection is. I have working template version now. Let me see how much I have to change to remove constness.

Reply:
I used `const_cast` to take away constness in the templated method.